### ISSUE TITLE
NEVER depend on ANY module inside the tree.

### ISSUE THREAD

**Issue by philderbeast** [ucsd-progsys/liquidhaskell#2065]
The module comment in `Language.Haskell.Liquid.GHC.TypeRep` is truncated:

https://github.com/ucsd-progsys/liquidhaskell/blob/c37ca0017f20070483ad0787e7f88f0223ac169a/src/Language/Haskell/Liquid/GHC/TypeRep.hs#L13-L22

It looks like the comment is a copy of a similar comment in `Language.Haskell.Liquid.GHC.Misc` but even that module doesn't heed the warning.

https://github.com/ucsd-progsys/liquidhaskell/blob/c37ca0017f20070483ad0787e7f88f0223ac169a/src/Language/Haskell/Liquid/GHC/Misc.hs#L17-L19

Could we either heed the warning or remove the warning?

**Comment by facundominguez** [comment]
Looks like the warnings make sense. The modules in `GHC/*` seem to be an interface that abstracts GHC versions.

I guess hlint could be used to enforce that modules in `GHC/*` do not import other modules from LH, and other modules in LH do not import from the ghc package directly.

**Comment by philderbeast** [comment]
The `GHC/*` modules are themselves within the `Language.Haskell.Liquid.*` tree so the warning is impossible to follow, as written when `GHC/*` modules depend on others within that subtree.

```
src> tree -L 4 -d
.
└── Language
    └── Haskell
        └── Liquid
            ├── Bare
            ├── Constraint
            ├── GHC
            ├── Interactive
            ├── Synthesize
            ├── Termination
            ├── Transforms
            ├── Types
            └── UX
```

**Comment by philderbeast** [comment]
Spinning off a `ghc-liquid` package (#1970) for modules in `Language.Haskell.Liquid.GHC` would prevent imports from going the wrong way.

**Comment by facundominguez** [comment]
> Spinning off a ghc-liquid package (#1970) for modules in Language.Haskell.Liquid.GHC would prevent imports from going the wrong way.

... or moving `Language.Haskell.Liquid.GHC` to `Language.Haskell.LiquidGHC`. I have no strong preference.

**Comment by facundominguez** [comment]
Seems to be addressed by #2067. Further steps could be discussed in a new issue.
